### PR TITLE
Use socket mode

### PR DESCRIPTION
**Pull request by harry-kim**
Wrapped the SocketModeClient so we could get the Flask app and use the context.
Might not be the best way, suggestions welcome.

### CODE REVIEW

File: nb2/bot/slack_bot.py [hunk 1]
```diff
@@ -34,8 +34,9 @@ def __init__(self):
         self.web_client = None
         self.slack_user_id = None
 
-    def init_app(self, token):
+    def init_app(self, token, app_token):
```
Comment (funkaoshi) [inline nb2/bot/slack_bot.py]: You aren't using app_token, should that be token.

Comment (harry-kim) [reply nb2/bot/slack_bot.py]: thanks removed

File: nb2/bot/socket_mode.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class FlaskSocketModeClient(SocketModeClient):
+    def __init__(self, app, app_token, web_client):
+        super().__init__(app_token=app_token, web_client=web_client)
+        self.app = app
+        self.socket_mode_request_listeners.append(self.process)
+        self.connect()
+
+    def process(self, client: SocketModeClient, req: SocketModeRequest):
+        if req.type == "events_api":
```
Comment (funkaoshi) [inline nb2/bot/socket_mode.py]: To get rid of the indenting, I normally write stuff like this as:

```
if req.type != "events_api":
    return

# Acknowledge the request anyway
....
```

Comment (harry-kim) [reply nb2/bot/socket_mode.py]: updated

File: nb2/bot/socket_mode.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    def process(self, client: SocketModeClient, req: SocketModeRequest):
+        if req.type != "events_api":
+            return
+
+        # Acknowledge the request anyway
+        response = SocketModeResponse(envelope_id=req.envelope_id)
+        client.send_socket_mode_response(response)
+
+        if (
+            req.payload["event"]["type"] == "message"
```
Comment (revangel) [inline nb2/bot/socket_mode.py:27]: I wonder if it's better we refactor all these conditionals into a method?

Comment (harry-kim) [reply nb2/bot/socket_mode.py:27]: can we add a ticket for that:P

**Comment by bellkelly** [comment]
+1

**Comment by revangel** [comment]
# Code (+1)

**Pull request merged**